### PR TITLE
fix(chat): stop false-positive stream recovery banner on new tasks

### PR DESCRIPTION
**Pull request by yiboyasss**
## Summary
- The periodic `task_stream_snapshot` reconciliation flagged any running task with no final-answer prefix seen yet as "interrupted", even though that is the normal state before streaming starts (planning, tool calls). Since the backend broadcasts this snapshot every 5s to every connected task, the recovery banner ("Live updates were interrupted...") ended up showing on almost every new task.
- Genuine misses are already detected elsewhere (explicit `stream_unavailable`/`stream_resync_required` messages, or a `final_answer_delta` arriving before its prefix), so this status-only guess is dropped rather than replaced.

## Test plan
- [x] Added a regression test reproducing the false positive (fails without the fix, passes with it)
- [x] `npx vitest run src/contexts/app-context-chat.test.tsx` — 186 passed
- [x] `npx tsc --noEmit` — no new errors in changed files
- [x] `npx eslint` on changed files — no new errors introduced

### CODE REVIEW

File: frontend/src/contexts/app-context-chat.tsx [hunk 1]
```diff
@@ -2816,10 +2816,26 @@ export function AppProvider({
           stream.runId = envelope.runId
           stream.prefixSeen = false
           stream.complete = false
+          // A new run starting can't have inherited an interruption from
+          // whatever run preceded it - carrying a stale `true` forward would
+          // flag the very first snapshot of a brand-new, healthy run.
+          // Mirrors the equivalent reset in the stream_run_id branch below.
+          stream.interrupted = false
```
Comment (rogercloud) [inline frontend/src/contexts/app-context-chat.tsx]: Major: this reset also fires on the very first snapshot processed for a freshly-created stream (`stream.runId` starts `undefined`), not only on a genuine run-to-run change. If `stream_unavailable`/`stream_resync_required` set `stream.interrupted = true` earlier (e.g. during task reload/reconnect while historical-data loading gated the snapshot branch), the next snapshot silently resets `interrupted` back to `false` within the ~5s cycle, with no real recovery having happened. This drops the recovery banner and lets durable snapshot output overwrite a completed message's content since the reducer gates that overwrite on `stream.interrupted`.

Fix: only reset on a defined-to-defined run change:
```ts
const isNewRun = stream.runId !== undefined && envelope.runId !== undefined && stream.runId !== envelope.runId
if (isNewRun) stream.interrupted = false
```

File: frontend/src/contexts/app-context-chat.tsx [hunk 2]
```diff
@@ -2816,10 +2816,26 @@ export function AppProvider({
...
         }
         stream.attemptId = typeof data.lease_attempt_id === "string" ? data.lease_attempt_id : null
         const active = envelope.status === "running"
```
Comment (rogercloud) [inline frontend/src/contexts/app-context-chat.tsx]: Minor: `active` is declared here but its only use (`dispatch({ type: "SET_PROCESSING", payload: active })`) is ~22 lines below, separated by an unrelated explanatory comment block about removing the old status-derived interrupted guess. Move this declaration next to its use, or shorten/relocate the comment.

File: frontend/src/contexts/app-context-chat.test.tsx [hunk 3]
```diff
@@ -745,6 +745,46 @@ describe("AppProvider websocket message routing", () => {
...
+    expect(screen.getByTestId("stream-recovery").textContent).toBe("")
+    send({ run_id: "run-1", state_version: 2, control_state: "running", status: "running", data: {} })
+    expect(screen.getByTestId("stream-recovery").textContent).toBe("")
+    send({ type: "final_answer_start", stream_run_id: "run-1", data: { message_id: "final_answer_1" } })
+    send({ type: "final_answer_delta", stream_run_id: "run-1", data: { message_id: "final_answer_1", delta: "on time answer" } })
+    expect(screen.getByTestId("messages").textContent).toContain("on time answer")
+    expect(screen.getByTestId("stream-recovery").textContent).toBe("")
+  })
+
+  it("clears recovery once a new run's snapshot supersedes an interrupted one", () => {
```
Comment (rogercloud) [inline frontend/src/contexts/app-context-chat.test.tsx:771]: Minor (test gap): this test establishes `run-1` via a snapshot first, so it only covers a genuine run-to-run transition, not the buggy ordering where `stream_unavailable`/`stream_resync_required` sets `interrupted = true` while `stream.runId` is still `undefined`, followed by the first-ever `task_stream_snapshot` for that stream. Add a test for that ordering to cover the fix for the `stream.runId !== envelope.runId` reset condition.

**Review by rogercloud — CHANGES_REQUESTED** [review]
### Major
- `frontend/src/contexts/app-context-chat.tsx:2823` — `stream.interrupted = false` fires on the very first snapshot for a fresh stream (`stream.runId` starts `undefined`), not only on a real run change, silently erasing a genuine interruption signal set by `stream_unavailable`/`stream_resync_required` within the ~5s snapshot cycle. Guard on a defined→defined transition:
```ts
const isNewRun = stream.runId !== undefined && envelope.runId !== undefined && stream.runId !== envelope.runId
if (isNewRun) stream.interrupted = false
```

### Minor
- `frontend/src/contexts/app-context-chat.tsx:2826` — `const active = envelope.status === "running"` is declared 12 lines away from its only use (`dispatch({ type: "SET_PROCESSING", payload: active })` at line 2848), separated by an unrelated explanatory comment block. Move the declaration next to its use, or shorten/relocate the comment.
- `frontend/src/contexts/app-context-chat.test.tsx:771` — the new test "clears recovery once a new run's snapshot supersedes an interrupted one" establishes `run-1` via a snapshot first, so it never exercises `stream_unavailable`/`stream_resync_required` arriving while `stream.runId` is still `undefined` followed by the first-ever snapshot. Add a test for that ordering alongside the fix above.

Blocking: yes — recommended event: REQUEST_CHANGES
frontend/src/contexts/app-context-chat.tsx:2823, major, silently clears a genuine stream-interruption flag on the first snapshot of a fresh stream, [new]

File: frontend/src/contexts/app-context-chat.tsx [hunk 4]
```diff
@@ -2812,14 +2812,36 @@ export function AppProvider({
...
         const data = asMessageRecord(message.data)
+        // A new run starting can't have inherited an interruption from
+        // whatever run preceded it - carrying a stale `true` forward would
+        // flag the very first snapshot of a brand-new, healthy run. But
+        // going from "no run known yet" to "this run" is NOT a run change:
+        // stream.runId starts undefined, and an explicit stream_unavailable/
+        // stream_resync_required for the CURRENT run can set interrupted
+        // before its first snapshot ever arrives - only a defined-to-
+        // different-defined transition is a genuine new run.
+        const isNewRun =
```
Comment (rogercloud) [inline frontend/src/contexts/app-context-chat.tsx:2823]: `stream.runId` is `string | null | undefined` and the backend sends `run_id: null` for a task that hasn't started running yet, with `extractTaskControlEnvelope` deliberately preserving `null` as distinct from `undefined`. So a `null -> "run-1"` transition is treated as a new run and clears `stream.interrupted` one snapshot early — the recovery banner disappears before anything confirms the stream is healthy (reachable because `stream_resync_required` fires on every (re)subscribe, including while the task is still pending).

Use `stream.runId != null && envelope.runId != null && stream.runId !== envelope.runId` so `null` gets the same treatment as `undefined`, and add a test covering the `run_id: null` -> real-run transition.

**Review by rogercloud — APPROVED** [review]
### Minor
- `frontend/src/contexts/app-context-chat.tsx:2823` — `isNewRun` only excludes `undefined`, so a `null -> "run-1"` transition (viewer attached to a pending task that then starts running) is treated as a genuine run change and clears `stream.interrupted` one snapshot early, dropping the recovery banner before the stream is known healthy. Extend the guard to exclude `null` too (`stream.runId != null && envelope.runId != null && stream.runId !== envelope.runId`), and add a test pinning the `run_id: null` -> real-run transition.

Blocking: no — recommended event: APPROVE

File: frontend/src/contexts/app-context-chat.tsx (hunk 4 above)
Comment (yiboyasss) [reply frontend/src/contexts/app-context-chat.tsx:2825]: Applied the suggested null-inclusive guard, but flagging a nuance I found while verifying: a real pending task's snapshot sending a flat top-level `run_id: null` doesn't actually reach `envelope.runId` as `null` — `extractTaskControlEnvelope`'s `root.run_id ?? task.run_id ?? data.run_id ?? nestedData.run_id` chain treats an explicit top-level `null` as nullish too, so it falls through the other (absent) fallbacks and resolves to `undefined`, not `null`. Confirmed with a quick script and by checking `task_stream_snapshot.py`, which always puts `run_id` at the top level. So the previous `undefined`-only guard was already safe for that exact case.

`envelope.runId` can only come out as a literal `null` through the innermost fallback (`nestedData.run_id`), which isn't how `task_stream_snapshot` messages are shaped today. Kept the broadened guard anyway since `runId` is typed `string | null` throughout and the change is free, and added a test that exercises it through that nested-data path rather than the flat shape, since that's the only way to genuinely reach `null` here.
